From cbd4258080058bf511c73032839060ff282849a3 Mon Sep 17 00:00:00 2001
From: Liu Hao <lh_mouse@126.com>
Date: Fri, 13 Apr 2018 20:51:04 +0800
Subject: [PATCH] Make GDB compile on MinGW targets again.

0) The GNU lib defines `open` and `close` as macros, which interfere
with member functions in the C++ standard library with the same names.
We have to `#undef` them before involving these C++ headers.

1) Add 'x86-tdep.o' into target object lists for i386 and amd64.
---
 gdb/configure.tgt                     | 4 ++--
 gdb/unittests/string_view-selftests.c | 3 +++
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/gdb/configure.tgt b/gdb/configure.tgt
index ba90411782..0b8b6c9f83 100644
--- a/gdb/configure.tgt
+++ b/gdb/configure.tgt
@@ -37,8 +37,8 @@ case $targ in
     ;;
 esac
 
-i386_tobjs="i386-tdep.o arch/i386.o i387-tdep.o"
-amd64_tobjs="amd64-tdep.o arch/amd64.o"
+i386_tobjs="i386-tdep.o x86-tdep.o arch/i386.o i387-tdep.o"
+amd64_tobjs="amd64-tdep.o x86-tdep.o arch/amd64.o"
 
 # Here are three sections to get a list of target specific object
 # files according to target triplet $TARG.
diff --git a/gdb/unittests/string_view-selftests.c b/gdb/unittests/string_view-selftests.c
index 182a5df9e4..7cdc2f9213 100644
--- a/gdb/unittests/string_view-selftests.c
+++ b/gdb/unittests/string_view-selftests.c
@@ -25,6 +25,9 @@
 #include "selftest.h"
 #include "common/gdb_string_view.h"
 
+#undef open
+#undef close
+
 /* Used by the included .cc files below.  Included here because the
    included test files are wrapped in a namespace.  */
 #include <string>
-- 
2.16.2

